Docs/Getting Started Guide.md

```diff
@@ -320,24 +311,36 @@ func fetchGitHubRepositories(state: State, store: Store<State>) -> Action? {
+In the example above we're using an `enum` to represent the different states of a single state slice that depends on a network request. It may look something like this:
+
+```swift
+enum GitHubFetchRequestState {
+    case Loading
+    case Repositories(Response<[Repository]>) // the OctoKit response provided to the `repositories` closure
+    case NetworkError(Error)
+}
+```
+
+There are many different ways to model states of a network request but it will mostly involve using multiple dispatched actions at different stages of your network requests.
 
 ### Action Creators
 
-An important aspect of adopting `ReSwift` is an improved separation of concerns. Specifically, your view layer should mostly be concerned with adopting its representation to match a new app state and for triggering `Action`s upon user interactions.
+An important aspect of adopting `ReSwift` is an improved separation of concerns. Specifically, your view controllers should mostly be concerned with adapting its representation to reflect a new app state and for triggering `Action`s upon user interactions.
 
-The triggering of actions should always be as simple as possible, we want to avoid any sort of complicated business logic in the view. However, in some cases it can be complicated to decide whether an action should be dispatched or not. Instead of checking the necessary state directly in the view or view controller, you can use `ActionCreator`s to perform a conditional dispatch.
+The triggering of actions should always be as simple as possible to avoid any sort of complicated business logic in the view controller. However, in some cases it can be complicated to decide whether an action should be dispatched or not. Instead of checking the necessary state directly in the view controller, you can use `ActionCreator`s to perform a conditional dispatch.
 
-Just like an `Action` a `ActionCreator` function can be dispatched to the store. An `ActionCreator` takes the current application state, and a reference to a store and might or might not return an `Action`.
+Just like an `Action`, an `ActionCreator` function can be dispatched to the store. An `ActionCreator` takes the current application state and a reference to a store then *may* return an `Action` (hence the `Action?` optional return type).
 
 An `ActionCreator` has the following type signature:
+
 ```swift
-typealias ActionCreator = (state: State, store: StoreType) -> Action?
+typealias ActionCreator = (_ state: State, _ store: StoreType) -> Action?
 ```
 
 A very simple example of an `ActionCreator` might be:
+
 ```swift
-func doubleValueIfSmall(state: TestAppState, store: Store<TestAppState>) -> Action? {
+func doubleValueIfSmall(_ state: TestAppState, _ store: Store<TestAppState>) -> Action? {
```

Comment:
Same as above -- even though the typealias cannot provide parameter names anymore, the implementation may, and thus add context on call sites.